src/messaging/ReliableMessageMgr.cpp

```diff
@@ -316,8 +316,9 @@ void ReliableMessageMgr::StartRetransmision(RetransTableEntry * entry)
     VerifyOrReturn(entry != nullptr && entry->rc != nullptr,
                    ChipLogError(ExchangeManager, "StartRetransmission was called for invalid entry"));
 
-    entry->nextRetransTimeTick = static_cast<uint16_t>(entry->rc->GetInitialRetransmitTimeoutTick() +
-                                                       GetTickCounterFromTimeDelta(System::Clock::GetMonotonicMilliseconds()));
+    entry->nextRetransTimeTick =
+        static_cast<uint16_t>(entry->rc->GetInitialRetransmitTimeoutTick() +
+                              GetTickCounterFromTimeDelta(System::SystemClock().GetMonotonicMilliseconds()));
```

Comment:
We are mixing "ticks" with milliseconds here as well

Reply:
Filed issue #10062 since these are well beyond the scope of the current PR.